api/blocks.js

```diff
@@ -1,44 +1,18 @@
 const blocks = require('../lib/api/blocks');
```

Comment:
you can write this file more clean and simple like this:
```javascript
const blocks = require('../lib/api/blocks');
const handler = require('./handler');

module.exports = function(app) {
    const blocksApi = new blocks(app);

    const serviceHandler = (req, res, next) => {
        let serviceName = req.params.service
        let serviceData
        switch (serviceName) {
            case 'getLastBlocks': serviceData = req.query.n; break;
            case 'getBlock': serviceData = req.query.blockId; break;
            case 'getHeight': serviceData = req.query.height; break;
        }
        handler(blocksApi, serviceName, serviceData, req, res, next)
    }

    app.get('/api/:service', serviceHandler)
}
```

Reply:
Although this approach eliminates many repetitive lines of our code, but it also introduces a problem: Since we use the same string to define the route (`/api/:service`) I think express will use the last registered `serviceHandler`.  Any suggestions?

Reply:
Ok, it a valid point I didn't think it though. A solution would be instead of registering route in every file. we define route configs as arrays. something like this:

routes.js:
```javascript

module.exports = [{
	path: '/getLastBlocks',
	handler: 'getLastBlocks',
	service: 'blocks',
	data: () => { /* pass required data */ }
}, {
	path: '/getBlock',
	handler: 'getBlock',
	service: 'blocks',
	data: () => { /* pass required data */ }
}, {
	path: '/getHeight',
	handler: 'getHeight',
	service: 'blocks',
	data: () => { /* pass required data */ }
}, {
	path: '/getBlockStatus',
	handler: 'getBlockStatus',
	service: 'blocks',
	data: () => { /* pass required data */ }
}]

```

then import those into index.js and register them dynamically:

```javascript

const routes = require('./routes.js')

routes.forEach(route => {
	serviceModule = require(`../lib/api/${route.service}`)
	service = new serviceModule(app)
	app.get(`/api/${route.path}`, (req, res, next) => {
		handler(service, route.handler, [PASS DATA HERE], req, res, next)
	})
})

```

Reply:
Great. Thanks. I used this approach with just a few tweaks to fit my case.